actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java

```diff
@@ -282,6 +283,37 @@ private void updateUser(Request actorMessage) {
         userMap, targetObject, correlatedObject, actorMessage.getContext());
   }
 
+  private void updateLocationCodeToIds(List<Map<String, String>> externalIds) {
+    List<String> locCodeLst = new ArrayList<>();
+    if (CollectionUtils.isNotEmpty(externalIds)) {
+      externalIds.forEach(
+          externalIdMap -> {
+            if (externalIdMap.containsValue(JsonKey.DECLARED_STATE)
+                || externalIdMap.containsValue(JsonKey.DECLARED_DISTRICT)) {
+              locCodeLst.add(externalIdMap.get(JsonKey.ID));
+            }
+          });
+      LocationClientImpl locationClient = new LocationClientImpl();
+      List<Location> locationIdList =
+          locationClient.getLocationByCodes(
+              getActorRef(LocationActorOperation.GET_RELATED_LOCATION_IDS.getValue()), locCodeLst);
+      if (CollectionUtils.isNotEmpty(locationIdList)) {
+        externalIds.forEach(
```

Comment:
Instead of iterating with externalIds, can you iterate with locaionIdList - I think this list will be smaller and there is no need to check state, district again. If code is present, just replace.
Pls test it outside in a simple java console app and this must be straightforward.

Reply:
done.